internal/characters/layla/shield.go

```diff
@@ -0,0 +1,209 @@
+package layla
+
+import (
+	"sort"
+
+	"github.com/genshinsim/gcsim/pkg/core/attributes"
+	"github.com/genshinsim/gcsim/pkg/core/combat"
+	"github.com/genshinsim/gcsim/pkg/core/glog"
+	"github.com/genshinsim/gcsim/pkg/core/player/shield"
+)
+
+const (
+	nightStars    = "nightstars"
+	starSkillIcd  = "nightstar-skill-icd"
+	starBurstIcd  = "nightstar-burst-icd"
+	shootingStars = "shooting-stars"
+)
+
+var starsTravel = []int{35, 33, 30, 28}
+
+type ICDNightStar int
+
+const (
+	ICDNightStarNone  ICDNightStar = iota
+	ICDNightStarSkill              // 0.3s
+	ICDNightStarBurst              // 0.5s
+)
+
+func (c *char) removeShield() {
+	if c.Tag(shootingStars) == 0 {
+		c.SetTag(nightStars, 0)
+	}
+	c.a1Stack = 0
+}
+
+func (c *char) newShield(base float64, dur int) *shd {
+	n := &shd{}
+	n.Tmpl = &shield.Tmpl{}
+	n.Tmpl.Src = c.Core.F
+	n.Tmpl.ShieldType = shield.ShieldLaylaSkill
+	n.Tmpl.Ele = attributes.Cryo
+	n.Tmpl.HP = base
+	n.Tmpl.Name = "Layla Skill"
+	n.Tmpl.Expires = c.Core.F + dur
+	n.c = c
+	return n
+}
+
+func (c *char) AddNightStars(count int, icd ICDNightStar) {
+	if c.Tag(shootingStars) > 0 {
+		return
+	}
+
+	switch icd {
+	case ICDNightStarSkill:
+		if c.StatusIsActive(starSkillIcd) {
+			return
+		}
+		c.AddStatus(starSkillIcd, 0.3*60, false)
+	case ICDNightStarBurst:
+		if c.StatusIsActive(starBurstIcd) {
+			return
+		}
+		c.AddStatus(starBurstIcd, 0.5*60, false)
+	}
+
+	if c.a1Stack < 4 {
+		c.a1Stack++
+	}
+
+	stars := count + c.Tag(nightStars)
+	if stars > 4 {
+		stars = 4
+	}
+	c.SetTag(nightStars, stars)
+	c.Core.Log.NewEvent("adding stars", glog.LogCharacterEvent, c.Index).
+		Write("stars", stars)
+
+	if stars == 4 && c.Tag(shootingStars) == 0 {
+		c.SetTag(shootingStars, 1)
+		c.shootStarSrc = c.Core.F
+		c.Core.Tasks.Add(c.ShootStars(c.shootStarSrc, -1), 0.5*60)
+	}
+}
+
+func (c *char) ShootStars(src int, last int) func() {
+	return func() {
+		if c.shootStarSrc != src {
+			return
+		}
+		if c.Tag(shootingStars) == 0 {
+			return
+		}
+
+		// find near target
+		nearTarget := -1
+		x, y := c.Core.Combat.Player().Pos()
+		trgs := c.Core.Combat.EnemiesWithinRadius(x, y, 10)
+		if len(trgs) > 0 {
+			sort.Slice(trgs, func(i, j int) bool { return i < j })
+			nearTarget = trgs[0]
+		}
+
+		if last == -1 {
+			// if not found
+			if nearTarget == -1 {
+				c.Core.Tasks.Add(c.ShootStars(src, -1), 0.1*60)
+				return
+			} else {
```

Comment:
Please reduce nesting by putting the stuff inside of the else right below the if statement and then removing the else.